tasks/upload.py

```diff
@@ -492,6 +493,7 @@ def run_impl_within_lock(
             self.retry(countdown=60, kwargs=kwargs)
         argument_list = []
 
+        name_creator = NameCreator()
```

Comment:
[nit] seems kinda random to instantiate `NameCreator` here - why not do this in the `if` statement where you need it, or earlier in the init method for this class?

Reply:
I ended up going with a possibly more simple pattern ([The global object pattern](https://python-patterns.guide/python/module-globals/)). I create a "default" instance in the `helpers.name_creator` and import that same instance to be used everywhere.